src/common/util/jsonPatcher.js

```diff
@@ -273,7 +328,7 @@ define([
 
     function _getChangedNodesRec(patch, res, hash, gmePath) {
         var nodePatches = patch[hash] && patch[hash].patch, // Changes regarding node with hash
-            i,
+            i, j,
```

Comment:
I am not quite sure what you mean by this comment.
If you think of generating an 'event' field for the patch object, then I believe it should be possible (though we still have to go through all the nodes in our territory to gather the events, but at least we could save the checking of every operation in a patch object...)